bulldozer/signals_test.go

```diff
@@ -137,7 +137,73 @@ func TestSignalsMatches(t *testing.T) {
 
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			matches, reason, err := signals.Matches(ctx, test.PullContext, "testlist")
+			matches, reason, err := signals.MatchesAny(ctx, test.PullContext, "testlist")
+			require.NoError(t, err)
+
+			if test.Matches {
+				assert.True(t, matches, "expected pull request to match, but it didn't")
+			} else {
+				assert.False(t, matches, "expected pull request to not match, but it did")
+			}
+			assert.Equal(t, test.Reason, reason)
+		})
+	}
+}
+
+func TestSignalsMatchesAll(t *testing.T) {
```

Comment:
Thanks for adding a test to cover the max commits behavior! I think this also needs to cover the "matches all" behavior, which is another important change.